### DISCUSSION TITLE
Welcome to autosarfactory Discussions!

### DISCUSSION THREAD

**Discussion by girishchandranc** [girishchandranc/autosarfactory#26] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by itsprajwalshavi21** [comment]
Hello!
Loved ur work.

I had a doubt.

#Read a list of files
files = ['some.arxml']
root, status = autosarfactory.read(files)

After this, can I perform operations on root and later export it as an ARXML??

Like adding a new AR_PACKAGE to the root and then export the generated ARXML??
Please let me know

**Reply by girishchandranc** [reply]
Hi @itsprajwalshavi21 ,
Thank you.

Sure, you can perform operations on the root and call autosarfactory.save() , which will update the relevant file.
Or. it's even possible to create a completely new file using the autosarfactory.new_file() and then add contents to it and save.

Please also bear in mind that this is a 1:1 representation of Autosar schema and hence will only allow you to add elements as specified by schema. For eg: Under the root, it's only possible to add AR-Package and under AR-Package, you can add any PackageableElement like ISignal, CompuMethod, System etc etc.

Hope it's clear for you.
